pycbc/events/stat.py

```diff
@@ -216,6 +255,196 @@ def coinc(self, s0, s1, slide, step): # pylint:disable=unused-argument
         return cstat
 
 
+class PhaseTDNewStatistic(NewSNRStatistic):
+    """Statistic that re-weights combined newsnr using coinc parameters.
+
+    The weighting is based on the PDF of time delays, phase differences and
+    amplitude ratios between triggers in different ifos.
+    """
+
+    def __init__(self, files, ifos=None):
+        NewSNRStatistic.__init__(self, files, ifos=ifos)
+
+        self.single_dtype = [('snglstat', numpy.float32),
+                             ('coa_phase', numpy.float32),
+                             ('end_time', numpy.float64),
+                             ('sigmasq', numpy.float32),
+                             ('snr', numpy.float32)
+                             ]
+
+        # Assign attribute so that it can be replaced with other functions
+        self.get_newsnr = ranking.get_newsnr
+        self.hist = self.hist_ifos = None
+        self.ref_snr = 5.0
+        self.relsense = {}
+        self.swidth = self.pwidth = self.twidth = None
+        self.max_penalty = None
+        self.pdtype = []
+        self.weights = {}
+        self.param_bin = {}
+
+    def get_hist(self, ifos=None):
+        """Read in a signal density file for the ifo combination"""
+
+        ifos = ifos or self.ifos
+
+        for name in self.files:
+            # Pick out the statistic files that provide phase / time/ amp
+            # relationships and match to the ifos in use
+            if 'phasetd_newsnr' in name:
+                ifokey = name.split('_')[2]
+                num = len(ifokey) / 2
+                if num != len(ifos):
+                    continue
+
+                match = [ifo in name for ifo in ifos]
+                if False in match:
+                    continue
+                else:
+                    break
+        else:
+            raise RuntimeError("Couldn't figure out which stat file to use")
+        logging.info("Using signal histogram %s for ifos %s", name, ifos)
```

Comment:
codeclimate complained about using 'name' outside the for loop .. I think it's probably best to avoid this - eg put 'selected = name' just before `break` ?